### ISSUE TITLE
Cannot reference both profileDOS and profileBotDefense. Duplicate profiles error message. 

### ISSUE THREAD

**Issue by megamattzilla** [F5Networks/f5-appsvcs-extension#273]
### Environment
 * Application Services Version: 3.19.1
 * BIG-IP Version: 15.1.0.2 

### Summary
Using AS3, I cannot create both a DoS profile and reference a Bot profile on Big-IP 15.1. When creating a DoS profile, AS3 seems to be creating a shadow f5_appsvcs_\<**DOS PROFILE NAME**\>_botDefense Bot profile which then creates a conflict when trying to add the desired Bot profile. 

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "class": "ADC",
    "schemaVersion": "3.19.0",
    "Common": {
        "class": "Tenant",
        "Shared": {
            "class": "Application",
            "template": "shared",
            "vs_http": {
                "layer4": "tcp",
                "pool": "http-pool",
                "translateServerAddress": true,
                "translateServerPort": true,
                "class": "Service_HTTP",
                "profileHTTP": {
                    "use": "/Common/Shared/custom-http"
                },
                "profileTCP": {
                    "use": "/Common/Shared/custom-tcp"
                },
                "profileDOS": {
                    "use": "/Common/Shared/custom-dos"
                 },
                 "profileBotDefense": {
                    "bigip": "/Common/bot-defense"
                },
                "virtualAddresses": [
                    "0.0.0.0/0"
                ],
                "virtualPort": 80,
                "snat": "auto"
            },
            "custom-http": {
                "trustXFF": true,
                "maxHeaderCount": 256,
                "maxHeaderSize": 65536,
                "hstsIncludeSubdomains": false,
                "proxyType": "reverse",
                "serverHeaderValue": "none",
                "class": "HTTP_Profile"
            },
            "custom-tcp": {
                "class": "TCP_Profile"
            },
            "http-pool": {
                "members": [
                    {
                        "addressDiscovery": "static",
                        "servicePort": 80,
                        "serverAddresses": [
                            "192.168.1.10"
                        ],
                        "shareNodes": true
                    }
                ],
                "monitors": [
                    {
                        "use": "/Common/Shared/http-monitor"
                    }
                ],
                "class": "Pool"
            },
            "custom-dos": {
                "application": {
                    "triggerIRule": true,
                    "captchaResponse": {
                        "failure": "You have entered an invalid answer for the question. Please, try again.\n<br>\n%DOSL7.captcha.image% %DOSL7.captcha.audio% %DOSL7.captcha.change%\n<br>\n<b>What code is in the image\\?</b>\n%DOSL7.captcha.solution%\n<br>\n%DOSL7.captcha.submit%\n<br>\n<br>\nYour support ID is: %DOSL7.captcha.support_id%.",
                        "first": "This question is for testing whether you are a human visitor and to prevent automated spam submission.\n<br>\n%DOSL7.captcha.image% %DOSL7.captcha.audio% %DOSL7.captcha.change%\n<br>\n<b>What code is in the image\\?</b>\n%DOSL7.captcha.solution%\n<br>\n%DOSL7.captcha.submit%\n<br>\n<br>\nYour support ID is: %DOSL7.captcha.support_id%."
                    },
                    "heavyURLProtection": {
                        "automaticDetectionEnabled": false
                    },
                    "rateBasedDetection": {
                        "thresholdsMode": "manual",
                        "escalationPeriod": 45,
                        "deEscalationPeriod": 600,
                        "sourceIP": {
                            "maximumTps": 40,
                            "minimumTps": 35
                        },
                        "url": {
                            "heavyURLProtectionEnabled": false,
                            "rateLimitingEnabled": false
                        }
                    },
                    "stressBasedDetection": {
                        "thresholdsMode": "manual",
                        "badActor": {
                            "mitigationMode": "none"
                        }
                    }
                },
                "class": "DOS_Profile"
            },
            "http-monitor": {
                "adaptive": false,
                "interval": 6,
                "dscp": 0,
                "receive": "OK$",
                "send": "GET /health HTTP/1.1\\r\\nConnection: Close\\r\\nHost: localhost\\r\\n\\r\\n",
                "timeUntilUp": 0,
                "timeout": 21,
                "class": "Monitor",
                "monitorType": "http",
                "targetAddress": "",
                "targetPort": 0
            }
        }
    }
}
```

2. Observe the following error response:
```json
{"results":[{"code":422,"message":"declaration failed","response":"01070097:3: Virtual server /Common/Shared/vs_http lists duplicate profiles.","host":"localhost","tenant":"Common","runTime":1948},{"code":422,"message":"declaration failed","response":"01070097:3: Virtual server /Common/Shared/vs_http lists duplicate profiles.","host":"localhost","tenant":"Common","runTime":2504}],"declaration":"HTML Tag-like Content in the Request URL/Body","code":422}
```

### Expected Behavior
A DoS profile named custom-dos to be attached to the virtual server. 
A Bot profile named bot-defense (default) attached to the virtual server. 

### Actual Behavior
The DoS profile named custom-dos is attached to the virtual server, along with a system-generated f5_appsvcs_custom-dos_botDefense bot profile. When AS3 tries to associate the specified bot profile bot-defense the error duplicate profiles error is observed. 

If I remove the profileDOS element from the declaration, I can apply the bot profile as expected. 

If I remove the profileBotDefense element from the declaration, I can apply the DoS profile as expected. 

I cannot reference both the profileDOS and profileBotDefense profiles due to the system generated bot profile from the DoS profile. 

For this example, I am using the Big-IP 15.1 bot-profile which TMSH config is: 
```security bot-defense profile bot-defense {
    app-service none
    whitelist {
        apple_touch_1 {
            match-order 2
            url /apple-touch-icon*.png
        }
        favicon_1 {
            match-order 1
            url /favicon.ico
        }
    }
}
```

**Comment by delgadillo22** [comment]
Thank for for your bug report. This is being tracked internally at https://jira.pdsea.f5net.com/browse/AUTOTOOL-1588

**Comment by dstokesf5** [comment]
This issue should be resolved in the 3.22.0 release.
